sdk/storage/azure-storage-blob/azure/storage/blob/_blob_client.py

```diff
@@ -283,10 +285,11 @@ def from_blob_url(
                     path_snapshot = snapshot['snapshot'] # type: ignore
                 except TypeError:
                     path_snapshot = snapshot
+        version_id = kwargs.pop('version_id', None)
```

Comment:
So you can technically just let this go through `kwargs` I would think instead of popping it here. This is true for connection string too.

Though here, we'll need to pop it when we end up parsing it from the URL so you could maybe leave it. Up to you.

Reply:
Just axed the popping here and will add it back in if we go down the URL parsing route!